### ISSUE TITLE
Installation problem on Ubuntu 17.10

### ISSUE THREAD

**Issue by pczekalski** [open-iscsi/open-iscsi#67]
Hi,

I cannot make installer work for me through the repository. The OS is Ubuntu 17.10.
For some unknown reasons, installer does not copy template config files.
Any help is appreciated.

`
pczekalski@n317-pc01:~$ sudo apt-get install open-iscsi 
[sudo] password for pczekalski: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libisns0
The following NEW packages will be installed:
  libisns0 open-iscsi
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/392 kB of archives.
After this operation, 2419 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Preconfiguring packages ...
Selecting previously unselected package libisns0:amd64.
(Reading database ... 240358 files and directories currently installed.)
Preparing to unpack .../libisns0_0.97-2_amd64.deb ...
Unpacking libisns0:amd64 (0.97-2) ...
Selecting previously unselected package open-iscsi.
Preparing to unpack .../open-iscsi_2.0.874-4ubuntu3_amd64.deb ...
Unpacking open-iscsi (2.0.874-4ubuntu3) ...
Processing triggers for ureadahead (0.100.0-20) ...
Setting up libisns0:amd64 (0.97-2) ...
Processing triggers for initramfs-tools (0.125ubuntu12) ...
update-initramfs: Generating /boot/initrd.img-4.13.0-16-generic
cp: cannot stat '/etc/iscsi/initiatorname.iscsi': No such file or directory
Setting up open-iscsi (2.0.874-4ubuntu3) ...
Job for iscsid.service failed because the control process exited with error code.
See "systemctl  status iscsid.service" and "journalctl  -xe" for details.
invoke-rc.d: initscript iscsid, action "restart" failed.
● iscsid.service - iSCSI initiator daemon (iscsid)
   Loaded: loaded (/lib/systemd/system/iscsid.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2017-11-22 15:46:40 CET; 6ms ago
     Docs: man:iscsid(8)
  Process: 10191 ExecStartPre=/lib/open-iscsi/startup-checks.sh (code=exited, status=1/FAILURE)

lis 22 15:46:40 n317-pc01 systemd[1]: Starting iSCSI initiator daemon (iscsid)...
lis 22 15:46:40 n317-pc01 startup-checks.sh[10191]: Error: configuration file /etc/iscsi/iscsid.conf is missing!
lis 22 15:46:40 n317-pc01 startup-checks.sh[10191]: The iSCSI driver has not been correctly installed and cannot start.
lis 22 15:46:40 n317-pc01 systemd[1]: iscsid.service: Control process exited, code=exited status=1
lis 22 15:46:40 n317-pc01 systemd[1]: Failed to start iSCSI initiator daemon (iscsid).
lis 22 15:46:40 n317-pc01 systemd[1]: iscsid.service: Unit entered failed state.
lis 22 15:46:40 n317-pc01 systemd[1]: iscsid.service: Failed with result 'exit-code'.
dpkg: error processing package open-iscsi (--configure):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for libc-bin (2.26-0ubuntu2) ...
Processing triggers for systemd (234-2ubuntu12.1) ...
Processing triggers for man-db (2.7.6.1-2) ...
Processing triggers for initramfs-tools (0.125ubuntu12) ...
update-initramfs: Generating /boot/initrd.img-4.13.0-16-generic
cp: cannot stat '/etc/iscsi/initiatorname.iscsi': No such file or directory
Errors were encountered while processing:
 open-iscsi
E: Sub-process /usr/bin/dpkg returned an error code (1)
`

`
pczekalski@n317-pc01:~$ systemctl status iscsid.service
● iscsid.service - iSCSI initiator daemon (iscsid)
   Loaded: loaded (/lib/systemd/system/iscsid.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2017-11-22 15:46:40 CET; 5min ago
     Docs: man:iscsid(8)

lis 22 15:46:40 n317-pc01 systemd[1]: Starting iSCSI initiator daemon (iscsid)...
lis 22 15:46:40 n317-pc01 startup-checks.sh[10191]: Error: configuration file /etc/iscsi/iscsid.conf is missing!
lis 22 15:46:40 n317-pc01 startup-checks.sh[10191]: The iSCSI driver has not been correctly installed and cannot start.
lis 22 15:46:40 n317-pc01 systemd[1]: iscsid.service: Control process exited, code=exited status=1
lis 22 15:46:40 n317-pc01 systemd[1]: Failed to start iSCSI initiator daemon (iscsid).
lis 22 15:46:40 n317-pc01 systemd[1]: iscsid.service: Unit entered failed state.
lis 22 15:46:40 n317-pc01 systemd[1]: iscsid.service: Failed with result 'exit-code'.
`

**Comment by gonzoleeman** [comment]
I downloaded the latest Ubuntu to test this out, and I got version 17.04. So not sure how you got 17.10. Beta?

Never the less, open-iscsi installs and works fine on 17.04. So please check with the Ubuntu folks, perhaps filing a bug there?

Disclaimer: I'm no expert on Debian packaging, as I use openSUSE/SLES.

**Comment by pczekalski** [comment]
Hi,

I'm sure there is 17.10 is officially released now. I'm using it for about 4 weeks already.
You can download it here:
https://www.ubuntu.com/download/desktop
2nd link (the first is for 16.04 LTS).

Regards,

P.

**Comment by gonzoleeman** [comment]
Ok, I downloaded 17.10 and I do not get your error.

Looking more closely at your output, it looks like this is the source of iscsid.service not running:

`● iscsid.service - iSCSI initiator daemon (iscsid)
Loaded: loaded (/lib/systemd/system/iscsid.service; enabled; vendor preset: enabled)
Active: failed (Result: exit-code) since Wed 2017-11-22 15:46:40 CET; 6ms ago
Docs: man:iscsid(8)
Process: 10191 ExecStartPre=/lib/open-iscsi/startup-checks.sh (code=exited, status=1/FAILURE)`

Looking at startup-checks.sh, it should print out an error message if it finds an issue. Try running it directly (using sudo, of course) to see what it is complaining about.

**Comment by pczekalski** [comment]
Hi,

Thanks for the quick reply. The error is: 

Error: configuration file /etc/iscsi/iscsid.conf is missing!

Surprisingly it is right after the fresh install of open-iscsi so I believe, iscsid.conf should be there already, during install, isn't it?

Anyway, I found something interesting - I've installed a fresh copy of U 17.10 on the virtual machine using new ISO downloaded from the Ubuntu website (same link as I provided to you above), installed open-iscsi and ... it just works with no issues. 

So I can blame my particular configuration running on my laptop. I suspect, one of the reasons is I've updated 17.04 to 17.10 (I had also some issues with iscsi daemon under 17.04 on the same machine). So I believe it is just my configuration that is the reason for non-installable open-scsi. I'm willing to make a fresh Ubuntu install from scratch so I'll let you know if this is a persistent issue or just one-time case.

Thank you for all your support and effort.

Regards,

P.
